hasher-matcher-actioner/hmalib/lambdas/api/api_root.py

```diff
@@ -261,31 +266,59 @@ def get_match_details(content_id: str) -> t.List[MatchDetailsResult]:
     records = PDQMatchRecord.get_from_content_id(
         table, f"{IMAGE_FOLDER_KEY}{content_id}"
     )
-    # TODO these mocked metadata should either be added to
-    # PDQMatchRecord or some other look up in the data model
-    mocked_metadata = MatchDetailsMetadata(
-        type="HASH_PDQ",
-        tags=["mocked_t1", "mocked_t2"],
-        status="MOCKED_STATUS",
-        opinions=["mocked_a1", "mocked_a2"],
-    )
-
-    mocked_actions = ["Mocked_False_Postive", "Mocked_Delete"]
     return [
         {
             "content_id": record.content_id[IMAGE_FOLDER_KEY_LEN:],
             "content_hash": record.content_hash,
             "signal_id": record.signal_id,
             "signal_hash": record.signal_hash,
             "signal_source": record.signal_source,
+            "signal_type": record.SIGNAL_TYPE,
             "updated_at": record.updated_at.isoformat(),
-            "meta_data": mocked_metadata,
-            "actions": mocked_actions,
+            "metadata": get_signal_details(record.signal_id, record.signal_source),
         }
         for record in records
     ]
 
 
+def get_signal_details(
+    signal_id: t.Union[str, int], signal_source: str
+) -> t.List[MatchDetailsMetadata]:
+    if not signal_id or not signal_source:
+        return []
+    table = dynamodb.Table(DYNAMODB_TABLE)
+    return [
+        MatchDetailsMetadata(
+            pg=metadata.pg_id,
+            tags=[
+                tag
+                for tag in metadata.tags
+                if tag
+                not in [
+                    ThreatDescriptor.TRUE_POSITIVE,
+                    ThreatDescriptor.FALSE_POSITIVE,
+                    ThreatDescriptor.DISPUTED,
+                ]
+            ],
+            opinion=get_opinion_from_tags(metadata.tags),
+        )
+        for metadata in PDQSignalMetadata.get_from_signal(
+            table, signal_id, signal_source
+        )
+    ]
+
+
+def get_opinion_from_tags(tags: t.List[str]):
```

Comment:
Can this return an enum instead? Or would that be too threat-exchange-y?

Also, are we sure all opinions are mutually exclusive? Can something be FALSE_POSITIVE and DISPUTED?

Reply:
Sure, you want an enum, I'll get you an enum :) 
(long term I want to refactor pytx but didn't want to be blocked by a pypi push/release right now)

Assuming mutually exclusive based on all the `elif` of  https://github.com/facebook/ThreatExchange/blob/master/python-threatexchange/threatexchange/descriptor.py#L78-L92
happy to program it more defensively though (the js just becomes a little rougher to handle)